packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3043,20 +3043,54 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.baseSnapshotBlobs = await SerializedSnapshotStorage.serializeTree(this.context.baseSnapshot, this.storage);
     }
 
+    private validatePendingMessages(pendingMessages: IPendingLocalState | undefined): void {
```

Comment:
Precisely because there is no scenario where it would be ok to return the pending state with an incomplete batch, I am adding this validation. This can be considered as extra validation, maybe not needed? @kian-thompson told me it could be nice to have it in this code path (getPendingLocalState).

Reply:
My point was that this check maybe belongs to the pending state manager

Reply:
Got it. Makes sense to move the entire method inside the pending state manager

Reply:
Yes. I am not sure about current state of whole batching model, but the reason behind adding this was the orderSequentially scenario in which I think this flush atomicity is not the case.  Even if we have already a check in getPendingLocalState that prevents it to be run while orderSequentially, this method was the insurance of it. Maybe it could also be an insurance for future scenarios and code paths as well.

Reply:
> the orderSequentially scenario in which I think this flush atomicity is not the case.

Do you have a repro?

Reply:
I think the `orderSequentially` scenario was something that happened before Vlad's changes around batching.
We're trying to get rid of all batching logic inside the PendingStateManager eventually, so I think this code should either live in ContainerRuntime or not at all (since it currently isn't possible for this validation to fail)